lib/resultuploaders/s3/s3.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+module AutoHCK
+  # S3 class for managing interactions with S3-compatible storage services.
+  # This class is designed to work with services that are compatible with the S3 API,
+  # including AWS S3 and Alibaba Cloud OSS.
+  class S3
+    CONFIG_JSON = 'lib/resultuploaders/s3/s3.json'
+    INDEX_FILE_NAME = 'index.html'
+
+    include Helper
+
+    # Provides read access to the URL for the index.html file.
+    #
+    # In the context of S3 compatibility, there is no concept of folders.
+    # The `url` will point to an index.html file that records the paths of all files.
+    # This allows `index.html` to be displayed directly in HTML format.
+    #
+    # Note: While AWS S3 supports this behavior of rendering HTML files directly,
+    # not all S3-compatible storage services do. For example, Alibaba Cloud OSS
+    # has security restrictions on its default domain that may require files to be
+    # downloaded rather than rendered directly, thus preventing HTML files from being displayed.
+    attr_reader :url
+
+    # rubocop:disable Metrics/AbcSize
+    def initialize(project)
+      @tag = project.engine_tag
+      @timestamp = project.timestamp
+      @logger = project.logger
+      @repo = project.config['repository']
+      @config = Json.read_json(CONFIG_JSON, @logger)
+
+      @access_key_id = ENV.fetch('AUTOHCK_S3_ACCESS_KEY_ID')
+      @secret_access_key = ENV.fetch('AUTOHCK_S3_SECRET_ACCESS_KEY')
+
+      @region = @config['region']
+      @bucket_name = @config['bucket_name']
+      @endpoint = @config['endpoint']
+
+      s3_resource = Aws::S3::Resource.new(
+        access_key_id: @access_key_id,
+        secret_access_key: @secret_access_key,
+        region: @region, endpoint: "https://#{@endpoint}"
+      )
+      @bucket = s3_resource.bucket(@bucket_name)
+
+      @index_template = ERB.new(File.read('lib/templates/index.html.erb'))
+      @index_obj = nil
+      @filenames = []
+    end
+    # rubocop:enable Metrics/AbcSize
```

Comment:
Let's not disable Metrics/AbcSize but split this method.
In particular, you can write a method that returns `Aws::S3::Bucket`. You can convert `@config`, `@access_key_id`, `@secret_access_key`, `@region`, `@bucket_name`, `@endpoint` to local variables of such a method.